pephubclient/schemas/schema.py

```diff
@@ -0,0 +1,445 @@
+            method="GET",
+            url=PEPHUB_SCHEMA_VERSIONS_URL.format(
+                namespace=namespace, schema_name=schema_name
+            ),
+            headers=self.parse_header(self.__jwt_data),
+            cookies=None,
+        )
+
+        if pephub_response.status_code == ResponseStatusCodes.OK:
+            decoded_response = self.decode_response(pephub_response, output_json=True)
+            return SchemaVersionResult(**decoded_response)
+
+        if pephub_response.status_code == ResponseStatusCodes.NOT_EXIST:
+            raise ResponseError("Schema doesn't exist, or you are unauthorized.")
+        if pephub_response.status_code == ResponseStatusCodes.INTERNAL_ERROR:
+            raise ResponseError(
+                f"Internal server error. Unexpected return value. Error: {pephub_response.status_code}"
+            )
+        else:
+            raise ResponseError(
+                f"Unexpected Status code return. Error: {pephub_response.status_code}"
+            )
+
+    def create_schema(
+        self,
+        namespace: str,
+        schema_name: str,
+        schema_value: dict,
+        version: str = "1.0.0",
+        description: str = None,
+        maintainers: str = None,
+        contributors: str = None,
+        release_notes: str = None,
+        tags: Union[str, List[str], dict, None] = None,
+        lifecycle_stage: str = None,
+        private: bool = False,
+    ) -> None:
+        """
+        Create a new schema record + version in the database
+
+        :param namespace: Namespace of the schema
+        :param schema_name: Name of the schema record
+        :param schema_value: Schema value itself in dict format
+        :param version: First version of the schema
+        :param description: Schema description
+        :param maintainers: Schema maintainers
+        :param contributors: Schema contributors of current version
+        :param release_notes: Release notes for current version
+        :param tags: Tags of the current version. Can be str, list[str], or dict
+        :param lifecycle_stage: Stage of the schema record
+        :param private: Weather project should be public or private. Default: False (public)
+
+        :raise: ResponseError if status not 202.
+        :return: None
+        """
+
+        url = PEPHUB_SCHEMA_NEW_SCHEMA_URL.format(namespace=namespace)
+        request_body = NewSchemaRecordModel(
+            schema_name=schema_name,
+            description=description,
+            maintainers=maintainers,
+            lifecycle_stage=lifecycle_stage,
+            private=private,
+            contributors=contributors,
+            release_notes=release_notes,
+            tags=tags,
+            version=version,
+            schema_value=schema_value,
+        ).model_dump(exclude_none=True)
+
+        pephub_response = self.send_request(
+            method="POST",
+            url=url,
+            headers=self.parse_header(self.__jwt_data),
```

Comment:
possible to add this to the `RequestManager` class so we aren't manually setting it everytime? Then you can choose to pass additional headers instead on each request?

Reply:
Maybe in the future:)